src/index.ts

```diff
@@ -1,34 +1,19 @@
 import { tick } from 'svelte';
 import { derived, writable } from 'svelte/store';
+import { prefs, sideEffect } from './shared';
 
 import { pathStore, queryStore } from './stores';
 
 interface SubmitEvent extends Event {
 	submitter: HTMLElement;
 }
 
-export interface Prefs {
-	query: {
-		array: {
-			separator: string
-			format: 'bracket' | 'separator'
-		}
-		nesting: number;
-		[key: string]: any;
-	};
-	sideEffect: boolean;
-}
-
 type HTMLFormControl = HTMLButtonElement & HTMLSelectElement & HTMLDataListElement & HTMLTextAreaElement & HTMLInputElement;
 
 const specialLinks = /((mailto:\w+)|(tel:\w+)).+/;
 
-const hasWindow = typeof window !== 'undefined',
-	hasHistory = typeof history !== 'undefined',
-	hasLocation = typeof location !== 'undefined',
-	hasProcess = typeof process !== 'undefined',
-	subWindow = hasWindow && window !== window.parent,
-	sideEffect = hasWindow && hasHistory && !subWindow;
+const hasLocation = typeof location !== 'undefined',
```

Comment:
Could you please move these lines to `./src/shared.ts` as well?

Reply:
Done.